DataStructures/BinarySearchTree/BinarySearchTree.cs

```diff
@@ -0,0 +1,275 @@
+                node.Right = Insert(value, node.Right);
+            }
+
+            // Returns the (unchanged) node as root of this subtree
+            return node;
+        }
+
+        /// <summary>
+        /// Function to search a given key in a given BST.
+        /// </summary>
+        /// <param name="value">Key value of type <c>T</c> you're searching for.</param>
+        /// <param name="node">Current node.</param>
+        /// <returns>Found node.</returns>
+        private Node<T>? Search(T value, Node<T>? node)
+        {
+            if (node is null)
+            {
+                // If node is null, the subtree is empty
+                // Returns an empty note, because it couldn't find the value
+                return node;
+            }
+
+            // Recur down the next subtree under consideration of the BTS properties
+            Comparer<T> defComp = Comparer<T>.Default;
+            if (defComp.Compare(value, node.Value) < 0)
+            {
+                // The value is smaller than the current node
+                return Search(value, node.Left);
+            }
+            else if (defComp.Compare(value, node.Value) > 0)
+            {
+                // The value is bigger than the current node
+                return Search(value, node.Right);
+            }
+            else
+            {
+                // If it isn't smaller or bigger it equals the current node
+                return node;
+            }
+        }
+
+        /// <summary>
+        /// Function deletes the key and returns the new root of this subtree.
+        /// </summary>
+        /// <param name="value">Key value of type <c>T</c> you're searching for.</param>
+        /// <param name="node">Current node.</param>
+        /// <returns>New root of this subtree.</returns>
+        private Node<T>? Remove(T value, Node<T>? node)
+        {
+            if (node is null)
+            {
+                // If node is null, the subtree is empty
+                // Returns an empty note, because it couldn't find the key
+                return node;
+            }
+
+            // Recur down the next subtree under consideration of the BTS properties
+            Comparer<T> defComp = Comparer<T>.Default;
+            if (defComp.Compare(value, node.Value) < 0)
+            {
+                // The value is smaller than the current node
+                node.Left = Remove(value, node.Left);
+            }
+            else if (defComp.Compare(value, node.Value) > 0)
+            {
+                // The value is bigger than the current node
+                node.Right = Remove(value, node.Right);
+            }
+            else
+            {
+                // If it isn't smaller or bigger it equals the current node
+
+                // Node with only one child
+                if (node.Left is null)
+                {
+                    return node.Right;
+                }
+                else if (node.Right is null)
```

Comment:
Codacy found an issue: [Add the missing 'else' clause.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6278013)